### PR TITLE
[Portfolio: preserve data in the browser] Microverse Project

### PR DESCRIPTION
**Pull request by mr-abdellah**
- Save data from contact form into local storage
- store the saved data in local storage and fill the form automatically.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -121,4 +123,33 @@ form.addEventListener('submit', (event) => {
...
+// Populate LocalStrorage
+function populateStorage() {
+  formObj = {
+    Name: form.fullname.value,
+    Email: form.email.value,
+    Message: form.message.value,
+  };
+  const convertObj = JSON.stringify(formObj);
+  localStorage.setItem('FormData', convertObj);
+}
```
Comment (Mre55) [inline script.js:136]: As required this project let's fix the `populateStorage` method currently it's setting the data to `local storage` when the user clicks and submits the form. But as stated in the project requirement it should have to set the data to `local storage` when the user changes any of the input fields 👇 .
image

If I fill out the form and before submitting it if I refresh the page I lose all the data I entered into the input field. Kindly, fix this to persist the data on page refresh even before submitting the form.

File: script.js [hunk 2]
```diff
@@ -121,4 +123,33 @@ form.addEventListener('submit', (event) => {
...
+  form.message.value = data.Message;
+}
+
+window.onload = function () {
+  form.user.addEventListener('input', populateStorage);
+  form.email.addEventListener('input', populateStorage);
+  form.message.addEventListener('input', populateStorage);
+  if (localStorage.getItem('FormData') !== null) {
+    getDataFromLocalStorage();
+  }
```
Comment (Mre55) [inline script.js]: Nice job 👍 here getting the local storage data on page load. Please fix the above issues I mentioned.

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @mr-abdellah,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._

**[REQUIRED]** Please update your PR description to be tell short summary of what you did in this PR.




## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 1 above)
Comment (danmainah) [reply script.js:136]: - [ ] Kindly check on your form input persistency as instructed by the reviewer, currently the input is lost on page reload.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @mr-abdellah, @caasperr, and @teckim, 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but some are still not addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Github flow was correctly used

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mr-abdellah** [comment]
> Hi @mr-abdellah, @caasperr, and @teckim,
> 
> Great work on making the changes requested by a previous reviewer 👏 👏 👏 You've done well implementing some of the requested changes, but some are still not addressed yet.
> 
> ## Highlights
> ✔️ No linter errors ✔️ Github flow was correctly used
> 
> # Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mr-abdellah** [comment]
Hello @danmainah , The function you metioned is not responsible for realtime data entry to localstorage. Realtime data entry is done in window.onload(); , please reply in comment because i still have one code review @danmainah

**Comment by danmainah** [comment]
@mr-abdellah  l will surely do.Its working kindly ask for another review.

**Pull request merged**